pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -83,15 +182,33 @@ void applyPauliY(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
                  const std::vector<std::size_t> &wires,
                  [[maybe_unused]] const bool inverse = false,
                  [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
-    applyNC1Functor(
-        ExecutionSpace{}, arr_, num_qubits, wires,
+    applyNCPauliY<ExecutionSpace, PrecisionT>(arr_, num_qubits, {}, {}, wires,
+                                              inverse);
+}
+
+template <class ExecutionSpace, class PrecisionT>
+void applyNCPauliZ(
+    Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
+    const std::size_t num_qubits,
+    const std::vector<std::size_t> &controlled_wires,
+    const std::vector<bool> &controlled_values,
+    const std::vector<std::size_t> &wires,
+    [[maybe_unused]] const bool inverse = false,
+    [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
+    auto core_function =
         KOKKOS_LAMBDA(Kokkos::View<Kokkos::complex<PrecisionT> *> arr,
                       const std::size_t i0, const std::size_t i1) {
-            const auto v0 = arr(i0);
-            const auto v1 = arr(i1);
-            arr(i0) = Kokkos::complex<PrecisionT>{imag(v1), -real(v1)};
-            arr(i1) = Kokkos::complex<PrecisionT>{-imag(v0), real(v0)};
-        });
+        [[maybe_unused]] const auto i0_ = i0;
```

Comment:
why unused here? do we need it?

Reply:
removed

Reply:
Actually after discussion with Ali, these are necessary for both GCC and Clang. These will be added back in later commit